### PR TITLE
Remove TODOs in `unused-variable` tests

### PR DESCRIPTION
**Pull request by jacobtylerwalls**
## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :hammer: Refactoring   |

## Description
Per discussion on #4391, which we closed, I think we can stick to the one-warning-per-variable-per-scope contract, so these TODOs are out of date.

If we want to look into adding warnings, we can do it more broadly in #5838.

### CODE REVIEW

File: tests/functional/u/unused/unused_variable.py [hunk 1]
```diff
@@ -66,7 +66,7 @@ def hello(arg):
         return True
     raise Exception
 
-# pylint: disable=redefined-outer-name, wrong-import-position,misplaced-future
+# pylint: disable=wrong-import-position,misplaced-future
```
Comment (jacobtylerwalls) [inline tests/functional/u/unused/unused_variable.py:69]: I thought it was worth surfacing the `redefined-outer-name` in these tests so you can more easily see that _some_ warning is raised for the fishiness.

**Review by Pierre-Sassoulas — APPROVED** [review]
Nice cleanup !

**Comment by jacobtylerwalls** [comment]
I'm looking into the behavior change -- see pylint run. Whether I have to restore the special case or not I'll add a functional test.

**Comment by jacobtylerwalls** [comment]
Hmm. Whatever it is belongs in another PR. Haven't narrowed it down yet. I'll restore the code I thought was useless, and then this can keep moving.

**Review by Pierre-Sassoulas — APPROVED** [review]

**Pull request merged**